test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ViewFeatures/DefaultHtmlGeneratorTest.cs

```diff
@@ -651,6 +652,26 @@ public void GetCurrentValues_WithModel_ReturnsModel(string[] rawValue, IReadOnly
             Assert.Equal(expectedAntiforgeryHtmlField, antiforgeryField);
         }
 
+        [Theory]
+        [InlineData(false)]
+        [InlineData(true)]
+        public void GenerateAntiforgery_AlwaysGeneratesAntiforgeryToken_IfCannotRenderAtEnd(bool hasAntiforgeryToken)
```

Comment:
Should this be `GenerateAntiforgery_AlwaysGeneratesAntiforgeryToken_WhenNotInsideForm`? Is this a valid scenario?